File: C#/第二套/11900-1060306/11900-1060306/Services/DataService.cs
namespace IdCardChecker.Services;

using IdCardChecker.Models;
using System.IO;

public interface IDataService
{
    Task<IEnumerable<IdCardRecord>> ReadRecordsAsync(string filePath);
}

public class DataService : IDataService
{
    public async Task<IEnumerable<IdCardRecord>> ReadRecordsAsync(string filePath)
    {
        if (!File.Exists(filePath))
        {
            return Enumerable.Empty<IdCardRecord>();
        }

        var lines = await File.ReadAllLinesAsync(filePath);
        return lines
            .Where(line => !string.IsNullOrWhiteSpace(line))
            .SelectMany(line =>
            {
                var parts = line.Split(',', StringSplitOptions.TrimEntries);
                if (parts.Length >= 3)
                {
                    return [new IdCardRecord(parts[0], parts[1], parts[2])];
                }
                return Enumerable.Empty<IdCardRecord>();
            });
    }
}

File: VB/第二套/11900-1060306/Services/DataService.vb
Imports System.IO

Namespace Services
    Public Interface IDataService
        Function LoadData(path As String) As IEnumerable(Of Models.IdCardRecord)
    End Interface

    Public Class DataService
        Implements IDataService

        Public Function LoadData(path As String) As IEnumerable(Of Models.IdCardRecord) Implements IDataService.LoadData
            Dim records As New List(Of Models.IdCardRecord)()
            If File.Exists(path) Then
                For Each line In File.ReadAllLines(path)
                    Dim ss = line.Split(","c)
                    If ss.Length >= 3 Then
                        records.Add(New Models.IdCardRecord(ss(0).Trim(), ss(1).Trim(), ss(2).Trim()))
                    End If
                Next
            End If
            Return records
        End Function
    End Class
End Namespace
